C++/include/algorithm/dynamic_programming/kadane.hpp

```diff
@@ -1,23 +1,41 @@
+#ifndef KADANE_HPP
+#define KADANE_HPP
+
 /*
-    Kadane's algorithm:
+    Kadane's Algorithm
     ------------------
     Used for finding the contiguous subarray within a one-dimensional array
     of integers which has the largest sum.
-    
+
     Time complexity
     ---------------
     O(N), where N is the number of elements in the original array.
-    
+
     Space complexity
     ----------------
-    O(1).
+    O(1)
 */
 
-#include <iostream>
 #include <tuple>
 #include <vector>
 
-using namespace std;
+using std::tuple;
+using std::make_tuple;
+using std::vector;
+
+/*
+    maximumSubarray
+    ---------------
+    Takes an array of integers as an argument and computes the maximum sum that
+    can be computed from any (contiguous) subarray of that array. Uses Kadane's
+    Algorithm, which employs dynamic programming.
+
+    Return value
+    ------------
+    tuple<int, size_t, size_t>, in which int is the maximum subarray value and
+    size_t to size_t represents the indices of the passed subarray that sum to
+    that value.
+*/
 
 tuple<int, size_t, size_t> maximumSubarray(const vector<int> &values)
```

Comment:
The function should be renamed to `maximum_subarray`. Also, we can have an alias for it named `kadane` like so:

```cpp
int f1(int x, int y) {
    // actual definition
    return x + y;
}

int f2(int x, int y) {
    // simple aliasing of f1 as f2
    return f1(x, y);
}
```
or,

```cpp
#define f2 f1   // f2 will now be replaced by f1, effectively acting as an alias

int f1(int x, int y) {
    return x + y;
}
```